WX-905

THWiseman opened the pull request:
This branch enables our integration tests to be run on the Github Actions CI platform.
- Added a Github Actions workflow file that will run integration tests 
- Added logic to a few of our bash functions in order to support a Github Actions environment
- 10 integration tests are included in this Github Action. They should all pass consistently. A few more tests remain un-ported from Travis, but they will come in a later branch.

jgainerdewar commented on .github/workflows/centaur_integration_tests.yml:
```diff
@@ -0,0 +1,61 @@
+#This github action runs all of Cromwell's integration tests.
+
+#This is what shows up in the github workflows page as the title. 
+run-name: ${{ github.actor }} Centaur Testing.
```
These aren't all Centaur test, right? Maybe should say Integration Testing instead?

jgainerdewar commented on .github/workflows/centaur_integration_tests.yml:
```diff
@@ -0,0 +1,61 @@
+      matrix:
+        include:
+          - build_type: centaurPapiV2beta
+            build_mysql: 5.7
```
It looks like this 5.7 gets included in the test name we see on the main PR page, but nothing makes it clear that it's describing mysql version. Is there an easy way to include that label?

THWiseman replied:
I added a "Friendly Name" field we can use for each test so we can see the more specific info from the Github UI :) 

The more elegant way of automagically using these include variables in the UI name isn't trivial at the moment because we're relying on multiple, different variables for the different databases we use.

jgainerdewar commented on .github/workflows/centaur_integration_tests.yml:
```diff
@@ -0,0 +1,61 @@
+  contents: read
+
+jobs:
+  centaur-tests:
```
As above, since these aren't all Centaur tests, can we use a different label here? Just `run-test` or something?

jgainerdewar commented on .github/workflows/centaur_integration_tests.yml:
```diff
@@ -0,0 +1,61 @@
+        cromwell_repo_token: ${{ secrets.BROADBOT_GITHUB_TOKEN }}
+    #This script bascially just looks up another script to run, assuming that the other script's filename is:
+    #src/ci/bin/test${BUILD_TYPE}.sh. The first letter of the BUILD_TYPE is automatically capitalized when looking.
+    - name: Run Centaur Test
```
What I said in the other two comments :)

jgainerdewar reviewed (commented):
This is exciting, nice work! Left a few comments just related to cleaning up the list of test names we see in the PR.

jgainerdewar reviewed (approved):
I love the friendly names! Assuming the currently-running tests pass this LGTM.

mcovarr commented on .github/workflows/integration_tests.yml:
```diff
@@ -0,0 +1,73 @@
+            friendly_name: DBMS
+          - build_type: centaurTes
+            build_mysql: 5.7
+            friendly_name: Centaur Tes with MySQL 5.7
```
nitpick alert!

```suggestion
            friendly_name: Centaur TES with MySQL 5.7
```

mcovarr commented on .github/workflows/integration_tests.yml:
```diff
@@ -0,0 +1,73 @@
+      BUILD_TYPE: ${{ matrix.build_type }} #intentionally duplicated variable
+      BUILD_MYSQL: ${{ matrix.build_mysql }}
+      BUILD_POSTGRESQL: ${{ matrix.build_postgresql }}
+      BUILD_MATRIADB: ${{ matrix.build_mariadb }}
```
typo, may lead to unexpected outcomes

```suggestion
      BUILD_MARIADB: ${{ matrix.build_mariadb }}
```

mcovarr reviewed (commented):
It's awesome to see this! 🤩

THWiseman commented:
I missed a one-liner that enabled these tests to work with databases other than mySQL when running on Github Actions. I added a line to include Github Actions, and added the `centaurLocal (mariaDB 10.3)`, `centaurLocal(postgreSQL 11.3)`, and `centaurEngingeUpgradeLocal(mySQL 5.7)` tests now that they work again. 

These tests are referenced in tickets WX-934 and WX-920.
